src/Adapter/Presenter/Module/ModulePresenter.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShop\PrestaShop\Adapter\Presenter\Module;
+
+use Currency;
+use PrestaShop\PrestaShop\Adapter\Product\PriceFormatter;
+use PrestaShop\PrestaShop\Adapter\Presenter\PresenterInterface;
+
+class ModulePresenter implements PresenterInterface
+{
+    /**
+     * @var Currency
+     */
+    private $currency;
+
+    /** @var PriceFormatter */
+    private $priceFormatter;
+
+    public function __construct(Currency $currency, PriceFormatter $priceFormatter)
+    {
+        $this->currency = $currency;
+        $this->priceFormatter  = $priceFormatter;
+    }
+
+    /**
+     * @param Module $module
+     *
+     * @return array
+     */
+    public function present($module)
+    {
+        if (!is_a($module, '\\PrestaShop\\PrestaShop\\Adapter\\Module\\Module')) {
```

Comment:
It's faster to use language construct `instanceof`, and use `Module::class` instead full string :)

Reply:
Otherwise use it in Parameter `present(Module $module)` it's better :)

Reply:
not possible because the presenterInterface is not typed

Reply:
So use `instanceof` instead, it will gain in perf :D